tests/test_util_typing.py

```diff
@@ -360,8 +360,8 @@ def test_stringify_type_hints_pep_585():
 
 def test_stringify_Annotated():
     from typing import Annotated  # type: ignore[attr-defined]
-    assert stringify_annotation(Annotated[str, "foo", "bar"], 'fully-qualified-except-typing') == "str"
-    assert stringify_annotation(Annotated[str, "foo", "bar"], "smart") == "str"
+    assert stringify_annotation(Annotated[str, "foo", "bar"], 'fully-qualified-except-typing') == "str[foo, bar]"
+    assert stringify_annotation(Annotated[str, "foo", "bar"], "smart") == "str[foo, bar]"
```

Comment:
Can you add some tests where the metadata is more than just strings but like real instances of objects and see what happens?  See https://peps.python.org/pep-0593/#combining-runtime-and-static-uses-of-annotations for some examples.